src/main/java/Duke.java

```diff
@@ -2,93 +2,166 @@
+                    taskCount = addTask(taskCount, ("[E] " + brokenDownAt[0] + "(at: " + brokenDownAt[1]) + ")");
+                    isValid = true;
                 }
-                isFinish = true;
             }
+            if(brokenDownInput[0].equals("invalid") || !isValid) {
+                System.out.println(INVALID_MESSAGE);
+            }
+            command = in.nextLine();
+            isValid = false;
+        }
+    }
+
+    private static int addTask(int taskCount, String command) {
+        taskCount++;
+        printTaskCount(taskCount, command);
+        return taskCount;
+    }
+
+    private static void printWelcome() {
+        System.out.println(LOGO);
+        printLine();
+        System.out.println("SQUAWK!!!");
+        System.out.println("How can I help you?");
+        printLine();
+    }
 
-            if(!command.isEmpty() && !isFinish) {
-                taskList[taskIndex] = new Task(command);
+    private static void listTask(Task[] tasks, int taskCount) {
+        if(taskCount > 0) {
+            Task[] taskList = Arrays.copyOf(tasks, taskCount);
+            int taskIndex = 0;
+            int index = 1;
+            printLine();
+            System.out.println(LISTING_MESSAGE);
+            printLine();
+            for(Task task: taskList) {
+                Task theTask = taskList[taskIndex];
+                theTask.printList(theTask,index);
+                index++;
                 taskIndex++;
-                printLine();
-                echoMessage(command);
-                printLine();
             }
-            command = in.nextLine();
-            isFinish = false;
+            printLine();
+        } else {
+            System.out.println("There are no task in the list!!");
         }
+    }
+
+    private static void markCompletion(Task[] tasks, String s) {
+        int taskLabel = Integer.parseInt(s);
+        int taskIndex = taskLabel - 1;
+        tasks[taskIndex].markDone();
+        printTaskCompletionMsg(taskLabel);
+    }
 
-        //If command is bye
+    private static void printTaskCompletionMsg(int taskNumber) {
         printLine();
-        System.out.println("SQUAWK! See you next time! :)");
+        System.out.println("Oooh I see you've done task " + taskNumber);
         printLine();
     }
+
+    private static void printTaskCount(int taskCount, String command) {
+        printLine();
+        echoMessage(command);
+        System.out.println("There are currently " + taskCount + " task now!");
+        printLine();
+    }
+
+    private static void printBye() {
+        printLine();
+        System.out.println(GREETING_MESSAGE);
+        printLine();
+    }
+    public static void printLine() {
+        for(int i = 0; i <= 50; i++) {
+            System.out.print("~");
+        }
+        System.out.println(" ");
+    }
+
+    public static void echoMessage(String line) {
+        System.out.println("Owl: I've added that!\nOwl: You added this: " + line);
+    }
 }
```

Comment:
Overall, decent job on the code, can include some comments here and there to help the reader keep up with your code but as a whole, I think its good and I don't really see much violations